invenio_rdm_migrator/streams/records/table_generator.py

```diff
@@ -90,7 +90,10 @@ def __init__(self, parent_cache, communities_cache):
                 RDMRecordFile,
             ],
             pks=[
-                ("record.id", generate_uuid),
+                (
+                    "record.id",
+                    lambda d: d["id"] if d.get("id") is not None else generate_uuid(d),
+                ),
```

Comment:
minor: an alternative, to avoid the weird lambda, could be to bake-in the logic in `TableGenerator._generate_pks`, like:

```python
    def _generate_pks(self, data):
        for path, pk_func in self.pks:
            if not dict_lookup(data, path):
                dict_set(data, path, pk_func(data))
```

Reply:
For files I needed to do some custom implementation to generate a pk for each file on the list so doing the above it would break that behaviour....I can extract it just to a function so we avoid the complicated lambda...